solr/core/src/java/org/apache/solr/cloud/overseer/ZkStateWriter.java

```diff
@@ -55,6 +56,8 @@
  */
 public class ZkStateWriter {
   private static final long MAX_FLUSH_INTERVAL = TimeUnit.NANOSECONDS.convert(Overseer.STATE_UPDATE_DELAY, TimeUnit.MILLISECONDS);
+  // If the state.json is greater than this many bytes and compression is enabled in solr.xml, then the data will be compressed
+  private static final int MIN_BYTES_FOR_COMPRESSION = 500000;
```

Comment:
@magibney Do you think there would be value in this being configured in solr.xml instead of a constant? Instead of configuring a boolean to enable compression, could configure a min bytes for compression.

Reply:
Yes, I do think that'd be valuable if it's not too much trouble to add. Even if we remove the configurability later, it'd be valuable to help find the "sweet spot" -- a lot of the benefit could be found on the "real-world zk" side, so it's possible that some experimentation may indicate that a smaller threshold actually yields benefits as well ...

Reply:
Maybe just expose java system property for now `zookpeer.minBytesForCompression`

Reply:
Done, we can set it as a system property in our solr.xml

Reply:
Nice, very clean :-)